Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskSet.cs

```diff
@@ -61,37 +66,56 @@ protected override void DisposeSelf()
             {
                 CancelRequestsContexts.Dispose();
             }
-            m_MigratableEntityPersistentDataByType.DisposeAllValuesAndClear();
 
             base.DisposeSelf();
         }
 
         public void AddResolvableDataStreamsTo(Type type, List<AbstractDataStream> dataStreams)
         {
-            if (!m_PublicDataStreamsByType.TryGetValue(type, out AbstractDataStream dataStream))
+            if (m_Debug_TypeCount.TryGetValue(type, out int count) && count > 1)
+            {
+                throw new InvalidOperationException($"Trying to get resolvable data streams for the type {type.GetReadableName()} but there are more than one. We don't support this at this time!");
+            }
+            
+            Type genericTypeArrayData = typeof(ActiveArrayData<>);
+            Type genericTypeEntityProxyWrapperType = typeof(EntityProxyInstanceWrapper<>);
+
+            Type specificEntityProxyWrapperType = genericTypeEntityProxyWrapperType.MakeGenericType(type);
+            Type specificArrayType = genericTypeArrayData.MakeGenericType(specificEntityProxyWrapperType);
+            
+            //This is temporary and works on magic constant typing based on the data inside to get the same ID. #241 should fix this.
+            DataTargetID targetID = AbstractData.GenerateWorldUniqueID(TaskSetOwner, specificArrayType, string.Empty);
+            
+            if (!m_DataStreamLookupByID.TryGetValue(targetID, out AbstractDataStream dataStream))
             {
                 return;
             }
 
             dataStreams.Add(dataStream);
         }
 
-        public EntityProxyDataStream<TInstance> GetOrCreateDataStream<TInstance>(CancelRequestBehaviour cancelRequestBehaviour)
+        public EntityProxyDataStream<TInstance> GetOrCreateDataStream<TInstance>(CancelRequestBehaviour cancelRequestBehaviour, string uniqueContextIdentifier)
```

Comment:
No default UID value?

Reply:
Correct, you can't get here without going through the public API in TaskDriverSystem which is where the default is set.